std/uni.d

```diff
@@ -8382,6 +8384,198 @@ unittest
     "HELLo"w.toUpperCase.equal("HELLO"d);
     "HELLo"d.toLowerCase.equal("hello"d);
     "HELLo"d.toUpperCase.equal("HELLO"d);
+
+    import std.utf : byChar;
+    assert(toLower("\u1Fe2") == toLowerCase("\u1Fe2").byChar.array);
+}
+
+import std.stdio;
+// generic capitalizer on whole range, returns range
+private auto toCapitalizer(alias indexFnUpper, uint maxIdxUpper, alias tableFnUpper,
+                           Range)(Range str)
+    // Accept range of dchar's
+    if (isInputRange!Range &&
+        isSomeChar!(ElementEncodingType!Range) &&
+        ElementEncodingType!Range.sizeof == dchar.sizeof)
```

Comment:
Done this way at Andrei's suggestion because it needs to allow enums whose base type is dchar, but not char or wchar which are implicitly convertible to dchar.

Note that the bulk of this code is copy/pasta from toCase.